File: lib/email/send.ts
import { Resend } from 'resend';

const FROM = process.env.EMAIL_FROM || 'DayMaxing <onboarding@resend.dev>';

// Lazily constructed so importing this module (e.g. during Next.js build-time
// page-data collection) doesn't require RESEND_API_KEY to be present.
let client: Resend | null = null;
function resend(): Resend {
  if (!client) client = new Resend(process.env.RESEND_API_KEY);
  return client;
}

export async function sendEmail(args: { to: string; subject: string; html: string }) {
  return resend().emails.send({
    from: FROM,
    to: args.to,
    subject: args.subject,
    html: args.html,
  });
}

File: api/app/email/send.py
"""Resend email send — port of lib/email/send.ts."""

from __future__ import annotations

import resend

from ..config import get_settings


def send_email(*, to: str, subject: str, html: str) -> dict:
    s = get_settings()
    resend.api_key = s.resend_api_key
    return resend.Emails.send(
        {"from": s.email_from, "to": to, "subject": subject, "html": html}
    )
